### PR TITLE
feat(plugin-chart-echarts): Radar chart POC

### PR DESCRIPTION
**Pull request by zhaoyongjie**
🏆 Enhancements

#### Added Radar Chart
- basic query panel
- color schema
- legend options
- labels options
- radar shape select
- support cross filter
- support each metric configuration

#### Additional
This PR dependence: https://github.com/apache/superset/pull/13983

#### Data Panel
image

#### Cross filter

https://user-images.githubusercontent.com/2016594/114886022-c75cbc00-9e39-11eb-89d3-90c12c8ae8ac.mp4

#### Config Metric

https://user-images.githubusercontent.com/2016594/114886257-fb37e180-9e39-11eb-9528-c1628723026f.mp4

### CODE REVIEW

File: plugins/plugin-chart-echarts/src/Radar/controlPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+      label: t('Query'),
+      expanded: true,
+      controlSetRows: [['groupby'], ['metrics'], ['adhoc_filters'], ['row_limit']],
+    },
+    {
+      label: t('Chart Options'),
+      expanded: true,
+      controlSetRows: [
+        ['color_scheme'],
+        // eslint-disable-next-line react/jsx-key
```
Comment (kgabryje) [inline plugins/plugin-chart-echarts/src/Radar/controlPanel.tsx:47]: Do we need to disable eslint here? I removed that line locally and it still works fine

File: plugins/plugin-chart-echarts/src/Radar/controlPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+          {
+            name: 'number_format',
+            config: {
+              type: 'SelectControl',
+              freeForm: true,
+              label: t('Number format'),
+              renderTrigger: true,
+              default: numberFormat,
+              choices: D3_FORMAT_OPTIONS,
+              description: `${t('D3 format syntax: https://github.com/d3/d3-format')} ${t(
```
Comment (kgabryje) [inline plugins/plugin-chart-echarts/src/Radar/controlPanel.tsx:106]: I think we should add a new line between the link to D3 and the next sentence

File: plugins/plugin-chart-echarts/src/Radar/controlPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        [<h1 className="section-header">{t('Radar')}</h1>],
+        [
+          {
+            name: 'is_circle',
+            config: {
+              type: 'CheckboxControl',
+              label: t('Circle radar shape'),
+              renderTrigger: true,
+              default: isCircle,
+              description: t('Whether to display the labels.'),
```
Comment (kgabryje) [inline plugins/plugin-chart-echarts/src/Radar/controlPanel.tsx:136]: The description here doesn't match the functionality

File: plugins/plugin-chart-echarts/src/Radar/index.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+   * advanced visualizations that require either post processing operations
+   * (pivoting, rolling aggregations, sorting etc) or submitting multiple queries.
+   */
+  constructor() {
+    super({
+      buildQuery,
+      controlPanel,
+      loadChart: () => import('./EchartsRadar'),
+      metadata: new ChartMetadata({
+        // behaviors: [Behavior.CROSS_FILTER],
```
Comment (kgabryje) [inline plugins/plugin-chart-echarts/src/Radar/index.ts:47]: Why is cross filtering disabled? It looks like the functionality is implemented

File: plugins/plugin-chart-echarts/src/Radar/controlPanel.tsx [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+        ],
+        [
+          {
+            name: 'label_type',
+            config: {
+              type: 'SelectControl',
+              label: t('Label Type'),
+              default: labelType,
+              renderTrigger: true,
+              choices: [
```
Comment (kgabryje) [inline plugins/plugin-chart-echarts/src/Radar/controlPanel.tsx:74]: Can we use `labelType` as one of the choices? The only way to go back to the default is to clear the select button. I think that selecting a choice would be more intuitive.

**Comment by kgabryje** [comment]
Can you please bump superset-ui-core and chart-controls versions, so that https://github.com/apache-superset/superset-ui/pull/1035 is included?

**Comment by zhaoyongjie** [comment]
> Can you please bump superset-ui-core and chart-controls versions, so that #1035 is included?

Hi @kgabryje,
I have been rebase master and added x-filter to Radar Chart.

File: plugins/plugin-chart-echarts/src/Radar/controlPanel.tsx (hunk 2 above)
Comment (zhaoyongjie) [reply plugins/plugin-chart-echarts/src/Radar/controlPanel.tsx:148]: In the control description looks that can not break lines.

Comment (kgabryje) [reply plugins/plugin-chart-echarts/src/Radar/controlPanel.tsx:161]: How about adding a dot at the end of first sentence?

Comment (kgabryje) [reply plugins/plugin-chart-echarts/src/Radar/controlPanel.tsx:161]:
```suggestion
              description: `${t('D3 format syntax: https://github.com/d3/d3-format.')} ${t(
```

**Review by kgabryje — DISMISSED** [review]
~LGTM! Pointed 1 _tiny_ nit, but it's good without it 🙂~
I noticed a bug - see comment below

**Review by kgabryje — COMMENTED** [review]
There's a bug in "Customize metrics" section. If you click on a value related to "group by", an exception is thrown (see video)

https://user-images.githubusercontent.com/15073128/114993975-995c9380-9e9c-11eb-9368-a5d5a17593f7.mov

**Review by kgabryje — APPROVED** [review]
LGTM!

**Pull request merged**